packages/react-combobox/src/components/Combobox/Combobox.types.ts

```diff
@@ -1,19 +1,96 @@
+import * as React from 'react';
+import type { PositioningShorthand } from '@fluentui/react-positioning';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import { OptionValue, OptionCollectionState } from '../../utils/OptionCollection.types';
+import { SelectionProps, SelectionState } from '../../utils/Selection.types';
+import { Listbox } from '../Listbox/Listbox';
+import { ComboButton } from '../ComboButton/ComboButton';
 
 export type ComboboxSlots = {
-  root: Slot<'div'>;
+  /* The root combobox slot */
+  root: NonNullable<Slot<'div'>>;
+
+  /* The dropdown listbox slot */
+  listbox: Slot<typeof Listbox>;
+
+  /* The primary slot, the element with role="combobox" */
+  trigger: Slot<typeof ComboButton>;
 };
 
 export type ComboboxCommons = {
-  // TODO Add things shared between props and state here
+  /**
+   * Controls the colors and borders of the combobox.
+   * @default 'outline'
+   */
+  appearance?: 'outline' | 'underline' | 'filledDarker' | 'filledLighter';
+
+  /**
+   * Render the combobox dropdown inline in the DOM.
+   * This has accessibility benefits, particularly for touch screen readers.
+   */
+  inline?: boolean;
+
+  /**
+   * Sets the open/closed state of the dropdown.
+   * Use together with onOpenChange to fully control the dropdown's visibility
+   */
+  open?: boolean;
+
+  /**
+   * If set, the placeholder will show when no value is selected
+   */
+  placeholder?: string;
+
+  /**
+   * The value displayed by the Combobox.
+   * Use this with `onSelect` to directly control the displayed value string
+   */
+  value?: string;
 };
 
 /**
  * Combobox Props
  */
-export type ComboboxProps = ComponentProps<ComboboxSlots> & ComboboxCommons;
+export type ComboboxProps = ComponentProps<Partial<ComboboxSlots>, 'trigger'> &
+  ComboboxCommons &
+  SelectionProps & {
+    /**
+     * Callback when the open/closed state of the dropdown changes
+     */
+    onOpenChange?: (e: OpenEvents, data: OnOpenChangeData) => void;
```

Comment:
```suggestion
    onOpenChange?: (e: ComboBoxOpenEvents, data: ComboBoxOpenChangeData) => void;
```

otherwise we will clash a lot of type namings